tuning/data/data_processors.py

```diff
@@ -282,46 +282,38 @@ def __execute_filter_data_handler(self, raw_datasets, handler, **kwargs):
             kwargs["desc"] = handler.desc
         return raw_datasets.filter(handler.op, **kwargs)
 
-    def __execute_map_data_handler(
-        self, raw_datasets, handler, splitName, datasetName, **kwargs
-    ):
-        column_names = None
-        if hasattr(raw_datasets[splitName], "column_names"):
-            column_names = raw_datasets[splitName].column_names
-
-        # remove __content__ from all processing
-        if column_names and "__content__" in column_names:
-            column_names.remove("__content__")
-
-        if "remove_columns" not in kwargs:
-            kwargs["remove_columns"] = None
-        if kwargs["remove_columns"] == "all":
-            if column_names is None:
-                logger.warning(
-                    "Could not infer column names from the dataset %s \
-                    unable to set `remove_columns` to all.\n \
-                    Please explicitly specify the column list or \
-                    use remove/select data handlers.",
-                    datasetName,
-                )
-            else:
-                kwargs["remove_columns"] = column_names
+    def __execute_map_data_handler(self, raw_datasets, handler, datasetName, **kwargs):
+        """Apply handler.op to all splits in raw_datasets instead of just a single split."""
+        processed_splits = {}
+        for split_name, split in raw_datasets.items():
+            column_names = getattr(split, "column_names", None)
+
+            if "remove_columns" not in kwargs:
+                kwargs["remove_columns"] = None
+            if kwargs["remove_columns"] == "all":
+                if column_names is None:
+                    logger.warning(
+                        "Could not infer column names from the split '%s' in '%s'. "
+                        "Unable to set `remove_columns` to all.\n"
+                        "Please explicitly specify the column list or use remove/select handlers.",
+                        split_name,
+                        datasetName,
+                    )
+                else:
+                    kwargs["remove_columns"] = column_names
 
-        if "fn_kwargs" not in kwargs:
-            kwargs["fn_kwargs"] = {}
+            if "fn_kwargs" not in kwargs:
+                kwargs["fn_kwargs"] = {}
+            kwargs["fn_kwargs"]["tokenizer"] = self.tokenizer
 
-        kwargs["fn_kwargs"]["tokenizer"] = self.tokenizer
-        kwargs["fn_kwargs"]["column_names"] = column_names
+            if not isinstance(raw_datasets, (IterableDatasetDict, IterableDataset)):
```

Comment:
This has to be a `IterableDataset` now as it cannot be `dict` in any scenario after you pull out the splits.

Reply:
the `raw_datasets` variable is not being changed during the loop, and it still can be dictionary!

Reply:
then we can base this check on `ds` I assume or create these few arguments once for all the splits as you won't have to again.